rfcs/syntax-type-alias-type-packs.md

```diff
@@ -38,13 +38,15 @@ type X<T...> = --
 type A<S...> = X<S...> -- T... = (S...)
 ```
 
-Similar to function calls, we want to be able to assign multiple regular types to a single type pack:
+Similar to function calls, we want to be able to assign zero or more regular types to a single type pack:
 ```lua
-type A = X                    -- T... = (). Note: check 'Alternatives'
+type A = X<>                  -- T... = ()
```

Comment:
I think that I would prefer to have this as `X<()>`.

Reply:
I feel like there's only two consistent syntaxes:

- When specifying type pack arguments, you must specify a type pack explicitly. This means that we don't automatically restructure input sequences, so `X<number, string>` fails when X accepts one type pack parameter
- When specifying type pack arguments, and type pack isn't explicitly specified, we take a sequence of types until the last type, or possibly until the next type pack (?), and convert that to a type pack.

In the latter case it feels like `X<>` should indeed work and produce an empty type pack parameter.

Reply:
I had the first option as an alternative in a draft, but I didn't even want to consider that:
```
One option that we have is to remove implicit pack assignment from a set of types
and a trailing tail and always require new explicit type pack syntax:

type X<T...> = --

type A<S...> = X<S...>        -- invalid
type B = X                    -- invalid
type C = X<number>            -- invalid
type D = X<number, string>    -- invalid
type E = X<...number>         -- invalid

type A<S...> = X<(S...)>      -- T... = (S...)
type B = X<()>                -- T... = ()
type C = X<(number)>          -- T... = (number)
type D = X<(number, string)>  -- T... = (number, string)
type E = X<(...number)>       -- T... = (...number)

But this doesn't feel natural and for variadic function argument lists, the syntax for
the argument tail is not special.
```